src/v/cloud_storage/recursive_directory_walker.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cloud_storage/logger.h"
+#include "seastarx.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/sstring.hh>
+
+#include <chrono>
+#include <memory>
+#include <utility>
+
+namespace cloud_storage {
+
+struct recursive_directory_walker {
```

Comment:
The instance of this type probably need method `stop` that will stop ongoing asynchronous `walk` call.  And this method should be called inside the `stop` method of the cache.

Reply:
Added `stop()` method